src/oc_erchef/apps/oc_chef_wm/src/chef_wm_search.erl

```diff
@@ -115,10 +115,22 @@ handle_undefined_start(undefined, Req) ->
 handle_undefined_start(Start, _Req) ->
     Start.
 
+% NOTE(Lincoln Baker):
+% http_uri:decode became deprecated, however there didn't appear to be a
+% direct drop-in library function to replace it for how we were using it.
+% uri_string:dissect_query/1 returns key-value pairs, and for our inputs
+% the value component will often or always be true based on my testing.
```

Comment:
But I'm also trying to track down the reason we approach this in this way.  In this case, we want to convert the value we receive for 'start' , which comes out of chef_index:search.  For all search providers, it's extracted from the response JSON - and the docs for both ES and solr state this value will be an int. 

For GET/POST requests, we hit this path when the value is for any reason `undefined` coming out of the search API call - in that case, we'll use the query value.  We pass the query value through this function to  convert it to an int, which then gets included in the search response body. 

It does look like the value of the query string could be URL encoded (webmachine docs aren't clear, and I didn't dive too deeply).  It is not possible to get the multi-key-value-pairs because we're not looking at a full query string - only at the resolved value. In addition, integers don't get any encoding treatment - so if the string doesn't directly convert to an int, it's not valid for our purposes. 

Separately, we hit it when we invoke `from_params` to build the query record (applies to both GET/POST), via `chef_index:query_from_params`, which does the same thing with validation and will raise if the value is non-integer. 

TL;DR:
- the replacement function does not work the same as the original, though it seems close. But we may not need this, because: 
  - the things we're calling unencode on are query string values, extracted via `wrq:get_qs_value`. 
  - these things are not full URL-encoded strings - they are a potentially URL-encoded portion of a value, not a series of KVPs 
  - any integer will not get url encoded; anything that does get url encoded is not a valid integer and can't be accepted. 

If that logic sounds sane and you agree, maybe we could simplify by  doing an int conversion of the original  string list in both places.

Reply:
Greetings.  Just to provide context which may or may not be useful, since I didn't have any information or context to go on when I wrote the replacement (including what this function was originally designed to do or why, what inputs it was designed to transform and the outputs it was supposed to generate, etc), I simply took all of the inputs that we use during integration tests (obtained by simply inserting `io:format` statements into appropriate locations), and made sure the replacement function I wrote gave the same outputs as those inputs.

Having said that, that may not be what we want :)

> If that logic sounds sane and you agree, maybe we could simplify by doing an int conversion of the original string list in both places.

If you could explain the above a little more (what you mean by int conversion, and what you mean by 'both places,' I'll be happy to get to work.

Reply:
EDIT:
A little more context.  IIRC, during testing, every value arriving into line 124's KVs (unencode function) was of the form `{"some-key", true}`, so that's basically what I designed the function to deal with.  I have no idea if that changes anything (or even if it should), I'm just throwing it out there.

Reply:
Ok, I made the changes you requested (if I understood you correctly).  Adhoc and umbrella are green (see comment below).